File: src/leetcode/python/medium/HIndex_274.py
class HIndex_274(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        l = len(citations)
        if l == 0:
            return 0
        h = [0] * (l + 1)
        for i in range(l):
            if citations[i] > l:
                h[l] += 1
            else:
                h[citations[i]] += 1
        cnt = 0
        for i in range(l, -1, -1):
            cnt += h[i]
            if cnt >= i:
                return i
        return 0

File: src/leetcode/java/HIndex_274.java
package leetcode.java;

public class HIndex_274 {
  public int hIndex(int[] citations) {
    int length = citations.length;
    if (length == 0) return 0;
    int[] h = new int[length + 1];
    for (int i = 0; i < length; i++) {
      if (citations[i] > length) h[length]++;
      else h[citations[i]]++;
    }
    int cnt = 0;
    for (int i = length; i >= 0; i--) {
      cnt += h[i];
      if (cnt >= i) return i;
    }
    return 0;
  }
}
